File: java/service/iothub-service-sdk/src/main/java/com/microsoft/azure/iot/service/sdk/FeedbackRecord.java
/*
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project root for full license information.
 */

package com.microsoft.azure.iot.service.sdk;

import java.time.Instant;

/**
 * Data structure for feedback record received
 * Provide getters and setters for feedback record batch properties
 */
public class FeedbackRecord
{
    private Instant enqueuedTimeUtc;

    public Instant getEnqueuedTimeUtc()
    {
        return enqueuedTimeUtc;
    }

    public void setEnqueuedTimeUtc(Instant enqueuedTimeUtc)
    {
        this.enqueuedTimeUtc = enqueuedTimeUtc;
    }

    private String originalMessageId;

    public String getOriginalMessageId()
    {
        return originalMessageId;
    }

    public void setOriginalMessageId(String originalMessageId)
    {
        this.originalMessageId = originalMessageId;
    }

    private String correlationId;

    public String getCorrelationId()
    {
        return correlationId;
    }

    public void setCorrelationId(String correlationId)
    {
        this.correlationId = correlationId;
    }

    private FeedbackStatusCode statusCode;

    public FeedbackStatusCode getStatusCode()
    {
        return statusCode;
    }

    public void setStatusCode(FeedbackStatusCode statusCode)
    {
        this.statusCode = statusCode;
    }

    private String description;

    public String getDescription()
    {
        return description;
    }

    public void setDescription(String description)
    {
        this.description = description;
    }

    private String deviceGenerationId;

    public String getDeviceGenerationId()
    {
        return deviceGenerationId;
    }

    public void setDeviceGenerationId(String deviceGenerationId)
    {
        this.deviceGenerationId = deviceGenerationId;
    }

    private String deviceId;

    public String getDeviceId()
    {
        return deviceId;
    }

    public void setDeviceId(String deviceId)
    {
        this.deviceId = deviceId;
    }

}

File: csharp/service/Microsoft.Azure.Devices/FeedbackRecord.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Microsoft.Azure.Devices
{
    using System;
    using Newtonsoft.Json;

    /// <summary>
    /// Contains properties of an individual feedback record. 
    /// </summary>
    public class FeedbackRecord
    {
        [JsonProperty(PropertyName = "originalMessageId", DefaultValueHandling = DefaultValueHandling.Ignore)]
        public string OriginalMessageId { get; set; }

        [JsonProperty(PropertyName = "description", DefaultValueHandling = DefaultValueHandling.Ignore)]
        public string Description { get; set; }

        [JsonProperty(PropertyName = "deviceGenerationId", DefaultValueHandling = DefaultValueHandling.Ignore)]
        public string DeviceGenerationId { get; set; }

        [JsonProperty(PropertyName = "deviceId", DefaultValueHandling = DefaultValueHandling.Ignore)]
        public string DeviceId { get; set; }

        [JsonProperty(PropertyName = "enqueuedTimeUtc", DefaultValueHandling = DefaultValueHandling.Ignore)]
        public DateTime EnqueuedTimeUtc { get; set; }

        [JsonProperty(PropertyName = "statusCode", DefaultValueHandling = DefaultValueHandling.Ignore)]
        public FeedbackStatusCode StatusCode { get; set; }
    }
}
